### ISSUE TITLE
Library version of SPSR

### ISSUE THREAD

**Issue by fwilliams** [mkazhdan/PoissonRecon#250]
It would be really nice to have some kind of library version of PoissonRecon. 

Instead of reading a PLY file, the user could perhaps provide an instance of a class which outputs vertices and normals as an argument to a reconstruction function in C++ code. This would make it easier to integrate SPSR into existing reconstruction pipelines where geometry is acquired programmatically and not necessarily saved to disk.

**Comment by starseeker** [comment]
Would https://github.com/mkazhdan/PoissonRecon/blob/master/Src/Reconstruction.example.cpp let you do what you need?  (I've been looking into this myself but haven't quite gotten it working properly yet.)

**Comment by mkazhdan** [comment]
Exactly. The main goal in providing the example was to show how one could interface into the code using the header files as a library.

**Comment by starseeker** [comment]
Update - here is a worked example of using this approach to integrate PoissonRecon into a third party code:  

https://github.com/BRL-CAD/brlcad/blob/main/src/libbg/spsr.cpp

We needed to add /bigobj when building on Windows (https://github.com/BRL-CAD/brlcad/blob/main/src/libbg/CMakeLists.txt#L52) and we needed a couple patches to PoissonRecon for MSVC and OSX:  

https://github.com/BRL-CAD/bext/blob/main/poissonrecon/poissonrecon_msvc.patch
https://github.com/BRL-CAD/bext/blob/main/poissonrecon/poissonrecon_osx.patch

We also added a few initializations and other warning quellage:

https://github.com/BRL-CAD/bext/blob/main/poissonrecon/poissonrecon_init.patch

@mkazhdan Would the above be worth submitting as pull requests?
